src/nvme/mi-mctp.c

```diff
@@ -487,13 +665,39 @@ nvme_mi_ep_t nvme_mi_open_mctp(nvme_root_t root, unsigned int netid, __u8 eid)
 		goto err_free_mctp;
 	}
 
+	mctp->resp_buf_aem_size = 4096;
+	mctp->resp_buf_aem = malloc(mctp->resp_buf_aem_size);
+	if (!mctp->resp_buf_aem) {
+		errno_save = errno;
+		goto err_free_rspbuf;
+	}
+
 	mctp->net = netid;
 	mctp->eid = eid;
 
 	mctp->sd = ops.socket(AF_MCTP, SOCK_DGRAM, 0);
 	if (mctp->sd < 0) {
 		errno_save = errno;
-		goto err_free_rspbuf;
+		goto err_free_aem_rspbuf;
+	}
+
+	mctp->sd_aem = ops.socket(AF_MCTP, SOCK_DGRAM, 0);
+	if (mctp->sd_aem < 0) {
+		errno_save = errno;
+		goto err_close_sd;
+	}
+
+	memset(&addr, 0, sizeof(addr));
+	addr.smctp_family = AF_MCTP;
+	addr.smctp_network = mctp->net;
+	addr.smctp_addr.s_addr = MCTP_ADDR_ANY;//mctp->eid;
+	addr.smctp_type = MCTP_TYPE_NVME | MCTP_TYPE_MIC;
+	addr.smctp_tag = MCTP_TAG_OWNER;
+
+	if (bind(mctp->sd_aem, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
```

Comment:
As mentioned in other comments, this will fail on the second invocation `nvme_mi_open_mctp`; you can only have one bind to the same local address. If we open the socket lazily, that may hide this error a little too. Will discuss separately though.